tests/llmcompressor/transformers/compression/test_run_compressed.py

```diff
@@ -3,77 +3,187 @@
+    uncompressed_model_stub = None
 
     @classmethod
     def setUpClass(cls):
         cls.test_dir = tempfile.mkdtemp()
 
-        # TODO: Give option on HFQuantizer to run run_compressed True/False
-        # currently hardcoded to True
-        cls.compressed_model = AutoModelForCausalLM.from_pretrained(
-            cls.model_stub,
+        quantization_config = CompressedTensorsConfig(run_compressed=False)
+
+        # Decompressed using HFQuantizer
+        # Linear foward
+        cls.decompressed_model = AutoModelForCausalLM.from_pretrained(
+            cls.compressed_model_stub,
             torch_dtype="auto",
             device_map="auto",
-            # run_compressed=True, # TODO: Give option on HFQuantizer
+            quantization_config=quantization_config,
         )
-        # TODO: Use ModelCompressor until decompression is supported through
-        # HFQuant/run_compressed can be turned off.
+
+        # Load model as is at the uncompressed state
+        # Linear forward
         cls.uncompressed_model = AutoModelForCausalLM.from_pretrained(
-            cls.empty_model,
-            torch_dtype=cls.compressed_model.dtype,
-            device_map=cls.compressed_model.device,
-        )
-        config = AutoConfig.from_pretrained(cls.model_stub)
-        compression_config = getattr(config, QUANTIZATION_CONFIG_NAME, None)
-        cls.compressor = ModelCompressor.from_compression_config(compression_config)
-        cls.compressor.quantization_config.quantization_status = (
-            QuantizationStatus.FROZEN
-        )
-        cls.compressor.decompress(
-            model_path=cls.model_stub, model=cls.uncompressed_model
+            cls.uncompressed_model_stub,
+            torch_dtype=cls.decompressed_model.dtype,
+            device_map=cls.decompressed_model.device,
         )
+        breakpoint()
 
-        cls.tokenizer = AutoTokenizer.from_pretrained(cls.model_stub)
+        cls.tokenizer = AutoTokenizer.from_pretrained(cls.compressed_model_stub)
 
-    def test_compressed_matches_uncompressed(self):
+    def test_compressed_matches_decompressed(self):
         SAMPLE_INPUT = [
             "I love 4-bit quantization because",
             "What is the capital of France?",
             "def fibonacci(n):",
         ]
 
+        decompressed_device = self.decompressed_model.device
+        uncompressed_device = self.uncompressed_model.device
+
+        # overwrite weights in cpu to cuda
+        self.decompressed_model = self.decompressed_model.to(decompressed_device)
+        self.uncompressed_model = self.uncompressed_model.to(uncompressed_device)
+
         inputs = self.tokenizer(SAMPLE_INPUT, return_tensors="pt", padding=True).to(
-            self.compressed_model.device
+            decompressed_device
         )
-        compressed_output = self.tokenizer.batch_decode(
-            self.compressed_model.generate(**inputs, max_length=50)
+
+        decompressed_output = self.tokenizer.batch_decode(
```

Comment:
Again, do not need to decode, only compare outputs

Reply:
this is fine to keep.

Reply:
Kyle you are correct. I want to keep it to check if the generations make sense. If it takes up too much run time or memory happy to remove